Source/DafnyLanguageServer/Workspace/TextDocumentLoader.cs

```diff
@@ -141,29 +148,54 @@ bool loadCanceled
       );
     }
 
-    public Task<DafnyDocument> VerifyAsync(DafnyDocument document, CancellationToken cancellationToken) {
+    public IObservable<DafnyDocument> Verify(DafnyDocument document, CancellationToken cancellationToken) {
+      notificationPublisher.SendStatusNotification(document.TextDocumentItem, CompilationStatus.VerificationStarted);
+      var progressReporter = new VerificationProgressReporter(document.TextDocumentItem, notificationPublisher);
+      var programErrorReporter = new DiagnosticErrorReporter(document.Uri);
+      document.Program.Reporter = programErrorReporter;
+      var implementationTasks = verifier.Verify(document.Program, progressReporter, cancellationToken);
+      foreach (var implementationTask in implementationTasks) {
+        implementationTask.Run();
+      }
 
-      return Task.Factory.StartNew(() => VerifyInternalAsync(document, cancellationToken), cancellationToken,
-        TaskCreationOptions.None, LargeStackScheduler).Unwrap();
+      var _ = NotifyStatusAsync(document.TextDocumentItem, implementationTasks);
```

Comment:
No await here?

Reply:
Nope, these notifications are not waited for. They can run on their own time. The `var _ = ` is to explicitly not use the `Task`.

Reply:
Ah, scary.  Thanks.